### PR TITLE
fix: improve refresh dbt button logic and sync UI with state

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #7736

### Description:

Set the loading state of `Refresh dbt button` once the user clicks it, ensuring more than 1 job gets triggered in very short time span; 
Syncs the loading state with: 
* User clicks on the button
* Job gets triggered ('started') and running ('running')
Loading state stops once job errors ('error') or is done('done')

> [!IMPORTANT]
> Also, we were invalidating the `tables/TABLE_ID` queries (so the /explores/:exploreUuid) on every success of `useJob`; Since we're refetching on a `500`ms interval, we should only invalidate `onSucess` + job status is `DONE`. This reduces calls to the network (even if cached)

Enabled state:
https://github.com/lightdash/lightdash/assets/7611706/a304c176-f4e2-41f1-a580-c83e8f4f31d0

Disabled state:
https://github.com/lightdash/lightdash/assets/7611706/5a74e301-871e-4f63-a241-f558cc949c93

Preview env state:
https://github.com/lightdash/lightdash/assets/7611706/e6e95d54-260d-43ac-b1fa-e05ceac87198

And with no permissions:
image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/RefreshDbtButton/index.tsx [hunk 1]
```diff
@@ -1,48 +1,84 @@
...
-            <Tooltip2
-                hoverCloseDelay={500}
-                interactionKind="hover"
-                content={
-                    <p>
+            <Popover withinPortal withArrow width={300}>
+                <Popover.Target>
+                    <Box
+                        sx={{
+                            cursor: 'pointer',
```
Comment (joaoviana) [inline packages/frontend/src/components/RefreshDbtButton/index.tsx:67]: so the user knows they can click the container but the button is disabled.

Comment (rephus) [reply packages/frontend/src/components/RefreshDbtButton/index.tsx:67]: I don't see the pointer cursor when the button is disabled :thinking:

File: packages/frontend/src/hooks/useRefreshServer.tsx [hunk 2]
```diff
@@ -109,7 +109,9 @@ export const useJob = (
                 ? false
                 : 500,
```
Comment (rephus) [inline packages/frontend/src/hooks/useRefreshServer.tsx:110]: We could reduce this frequency

**Review by rephus — APPROVED** [review]
Tested, works as expected :ok_hand:

File: packages/frontend/src/components/RefreshDbtButton/index.tsx (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/components/RefreshDbtButton/index.tsx:67]: oh! Checking!

Comment (joaoviana) [reply packages/frontend/src/components/RefreshDbtButton/index.tsx:67]: https://github.com/lightdash/lightdash/assets/7611706/4fec0583-141c-42b9-8980-9813cfa8811b

interesting ^ I still see it...

**Pull request merged**